content/articles/2019-03-ggplot2-3-2-0.Rmarkdown

```diff
@@ -0,0 +1,160 @@
+primitives to use, and it takes care of the details.
+
+The 3.2.0 release is a minor release with focus on performance improvements, but
+also includes a few new features and a range of bug fixes. There are no breaking
+changes in this release, though a few changes may affect the visual appearance 
+of plots (noted below). Further, there are a few changes that developers of
+extension packages may need to take into account, but won't affect regular 
+users. For a full overview of the changes, please see the 
+[release notes](https://ggplot2.tidyverse.org/news/index.html).
+
+This release also includes a range of contributions made during our tidyverse
+developer day in Austin, many from first-time contributors. We hope they've
+become inspired to continue taking part in the development.
+
+Lastly, this release also sees the entrance of Hiroaki Yutani into the core
+developer team. Hiroaki has been amazing in tackling small and large issues both
+before and after he joined the team, and we are very lucky to have him.
+
+## Performance
+A large part of this release relates to internal changes intended to speed up
+ggplot2. Most of these are general and will affect the rendering speed of all
+the plots you make, but there has also been a specific focus on `geom_sf()`. All
+of this is part of a larger effort to make plotting in R more performant and 
+also include changes in [gtable](https://gtable.r-lib.org), 
+[sf](http://r-spatial.github.io/sf/), and R itself. Make sure to use the latest
+version of all of these packages to get the full benefit of the improvements.
+
+### geom_sf
+While most changes are general, `geom_sf()` has received special attention, as 
+it's performance have been less stellar compared to the rest of ggplot2. Most of
+this comes down to how `geom_sf()` converted its input to grobs, which was done
+row by row. This meant that plotting 10.000 `ST_POINT` objects would create 
+10.000 point grobs instead of a single grob containing 10.000 points. The same
+was true for all other data types. With the new release ggplot2 will try to 
+pack all object into a single grob, which is possible if all objects are of the
+same type (`MULTI*` types can be mixed with scalar types). Packing polygons into
+a single grob will only be possible when the next release of R is available, but
+all other types are otherwise supported. If the data contains a mix of types 
+ggplot2 will fall back to creating a grob per row, but this is a much less 
+frequent situation. The other big performance bottleneck in plotting sf data was
+the normalization of the coordinates. As sf stores its data in nested lists the
+standard vectorization in R doesn't apply, which led to much worse performance
+compared to normalizing data stored on standard data.frame format. The new 
+release of sf includes optimized functions for these operations implemented in C
+so plotting performance has improved immensely.
```

Comment:
Can we include some numbers here?

Reply:
I'm reluctant because it will vary from user to user and plot to plot, so it will be a surefire way to get people to complain about over-promising stuff